Major change. Restructured many components

jkandasa opened the pull request:
* Rearrange more organized way.
* Removed [widgetastic.patternfly](https://github.com/RedHatQE/widgetastic.patternfly) dependency
* Added logger
* Smart configuration (takes env and replaces with env.yaml)
* more fixtures
* etc.,
* not updated the docker stuff

To run this changes on your machine,
```
cd kiali-qe-python/
virtualenv .env
source .env/bin/activate
# update conf/emv.yaml (kiali hostname and selenium driver url)
pytest -s
# see the log on log/kiali_qe.log
```

jkandasa commented:
kialiqeci

hhovsepy commented:
@jkandasa please return back kiali rest api calls and existing tests

hhovsepy commented on conf/env.yaml:
```diff
@@ -1,9 +1,35 @@
-        unexpectedAlertBehaviour: 'ignore'
+# kiali instance details
+kiali:
+  hostname: localhost
```
keep this format: ${kiali_hostname} it's a key for our scripts

jkandasa replied:
@hhovsepy I have added support to update `env` dynamically with env.yaml. No need to update via script. If we want to update something from from environment, just add a map I have mentioned in the comment.

hhovsepy commented on kiali_qe/tests/test_rules.py:
```diff
@@ -1,30 +0,0 @@
-from project.kiali_view import RuleListView
-
-
-def test_rule_list(browser, rest_api):
```
existing tests should not suffer

jkandasa replied:
@hhovsepy These test are partially covered by `kiali_qe/tests/test_istio_mixer_page.py`. Yet to add the code to validate content. At this time it is validating only the count.

hhovsepy replied:
ok looking forward for content validation

hhovsepy commented on kiali_qe/tests/test_services.py:
```diff
@@ -1,28 +0,0 @@
-from project.kiali_view import ServiceListView
-
-
-def test_service_list(browser, rest_api):
```
existing tests should not suffer

jkandasa replied:
@hhovsepy These test are partially covered by `kiali_qe/tests/test_services_page.py`. Yet to add the code to validate content. At this time it is validating only the count.

hhovsepy commented on requirements.txt:
```diff
@@ -1,22 +1,11 @@
-widgetastic.core
-widgetastic.patternfly
-six
-pytest_jira
```
please return back this dependency

jkandasa replied:
@hhovsepy I have added supported dependency. And removed Which is not used in the framework. Some of dependency will be add by another dependency. Kindly let me know which dependency you want to add me from the removed list.

hhovsepy replied:
please add 'pytest_jira', even it is not used now, we could need it, because of future jira issues

jkandasa replied:
@hhovsepy ok

jkandasa commented on kiali_qe/fixtures/configuration.py:
```diff
@@ -0,0 +1,28 @@
+        'selenium.capabilities.platform': 'SELENIUM_PLATFORM',
+        'selenium.capabilities.browser': 'SELENIUM_BROWESR',
+        'selenium.capabilities.zalenium.recordVideo': 'ZAL_RECORD_VIDEO',
+        'selenium.capabilities.zalenium.idleTimeout': 'ZAL_IDLE_TIMEOUT'
```
@hhovsepy Here is the mapped environment variable and env.yaml file content location. If value not found in environment just keeps env.yaml value.

hhovsepy replied:
ok thank you

kialiqeci commented:
Build comment file: 
--------------
:red_circle: Jenkins CI: python-ui-test-pr [#33](https://paste.fedoraproject.org/paste/9sCxIwhpDfdfKU9zAzTufQ)
  * :red_circle: run-python-ui-tests [#252](https://paste.fedoraproject.org/paste/wqx2sV5QdDQ0tMPOrGPZ9Q)
--------------

kialiqeci commented:
Build comment file: 
--------------
:red_circle: Jenkins CI: python-ui-test-pr [#34](https://paste.fedoraproject.org/paste/dfewtAZptvIbSEZoRfhJSQ)
  * :red_circle: run-python-ui-tests [#265](https://paste.fedoraproject.org/paste/4~n8vWnqmN5gaOWybR12cw)
--------------

hhovsepy commented:
PR is run locally, all tests passing.

mattmahoneyrh commented:
All tests passed localhost - I'll be it very slowly.

setup.sh need to be clean up:
 - setuptools and pip need to be updated prior to requirements.txt module install
 - pycurl, oc, recorder, and moviepy apparently are not used by this project. Thus remove.

kialiqeci commented:
Build comment file: 
--------------
:red_circle: Jenkins CI: python-ui-test-pr [#37](https://paste.fedoraproject.org/paste/yr1iaWpNBFT0MAAgmqjwkQ)
  * :red_circle: run-python-ui-tests [#280](https://paste.fedoraproject.org/paste/Jl7SHvNRtJu6nWjoZrm8Iw)
--------------

kialiqeci commented:
Build comment file: 
--------------
:red_circle: Jenkins CI: python-ui-test-pr [#42](https://paste.fedoraproject.org/paste/CgsL1IXTFM7orvtrSL7WYA)
  * :red_circle: run-python-ui-tests [#320](https://paste.fedoraproject.org/paste/tL-vC02UvcBI72tNslZF1A)
--------------

kialiqeci commented:
Build comment file: 
--------------
:red_circle: Jenkins CI: python-ui-test-pr [#43](https://paste.fedoraproject.org/paste/KibYclZIVBAqw3Fiuxt76A)
  * :red_circle: run-python-ui-tests [#321](https://paste.fedoraproject.org/paste/SWmPgKSNqMmH9ImS25uOaA)
--------------

kialiqeci commented:
Build comment file: 
--------------
:red_circle: Jenkins CI: python-ui-test-pr [#44](https://paste.fedoraproject.org/paste/qhds1pD89Esc5yYvHrazwA)
  * :red_circle: run-python-ui-tests [#324](https://paste.fedoraproject.org/paste/kYGf8-vlxlOtS4vVkKlyyw)
--------------

jkandasa commented:
@hhovsepy @mattmahoneyrh Thank you!

@FilipB I have updated `Dockerfile` and removed `setup.sh` which is not required. I have created temporary jenkins slave image in our openshift and added it in our jenkins2.
Jenkins job: `run-python-ui-tests-modified`
Jenkins slave image: `jenkins-slave-kiali-ui-tests-modified`
You can also watch the zalenium recordings.

FilipB commented:
Thanks @jkandasa

kialiqeci commented:
Build comment file: 
--------------
:red_circle: Jenkins CI: python-ui-test-pr [#45](https://paste.fedoraproject.org/paste/7P7I6mU2BZ37ewmxRqwnSQ)
  * :red_circle: run-python-ui-tests [#330](https://paste.fedoraproject.org/paste/OhktdYpcqs9DkW1eg34tRQ)
--------------

jkandasa commented:
@hhovsepy @mattmahoneyrh @gbaufake I have updated all your review comments. Kindly review it and approve. If you feel everything looks good.

hhovsepy commented:
existing tests were removed, but it does not make sense to argue, so merging and continue
